[Ivor] iP

ivorcmx opened the pull request:
### _**DukeGenie Bot**_

>  "Are you looking at me? Did you rub my lamp? Did you wake me up? Did you bring me here? And all of a sudden you're walking out on me? I don't think so. Not right now! YOU'RE GETTIN' YOUR WISHES, SO *SIDDOWN!*" - Genie, Aladdin

:genie_man::genie_man::genie_man::genie_man::genie_man::genie_man:

DukeGenie frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _Rub the lamp and wish away_

All you need to do is,

1. Download it from [here](https://github.com/ivorcmx/ip)
2. Double-click it
3. Add your wishes
4. See your wish get granted :genie_man:

Features:

- [x] Remembering Tasks
- [x] Save and restore Tasks
- [ ] Managing Deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are genie as well, you can use it to practice your wishes too. Here's the `main` method:
```ruby
public static void main(String[] args) throws IOException {
        Ui.run();
    }

Commit:
Add Gradle support

ivorcmx committed:
Greet, Echo, Exit

ivorcmx committed:
Level-1

ivorcmx committed:
Add,List

ivorcmx committed:
Add,List

ivorcmx committed:
Revert "Add,List"

This reverts commit e19f843b593589e435567d84a3889a43f0bc10bf.

ivorcmx committed:
Merge branch 'master' of https://github.com/ivorcmx/iP

ivorcmx committed:
Mark as Done

ivorcmx committed:
ToDo, Event, Deadline

ivorcmx committed:
Merge branch 'master' of https://github.com/ivorcmx/iP

ivorcmx committed:
Automated Text UI Testing

ivorcmx committed:
Error Handling

ivorcmx committed:
Delete tasks

ivorcmx committed:
Moved print of greetings into a method

ivorcmx committed:
Moved greeting messages to methods

ivorcmx committed:
Created class to print tasklists

ivorcmx committed:
Added unmark and mark methods

ivorcmx committed:
Added addDeadline method

ivorcmx committed:
Added todo and event method

ivorcmx committed:
Added delete method

ivorcmx committed:
Added type for task

ivorcmx committed:
Added eraselist

ivorcmx committed:
Added save function

ivorcmx committed:
Add date format for deadline

ivorcmx committed:
A-MoreOOP

ivorcmx committed:
Fixed bye command

ivorcmx committed:
Added duke package

ivorcmx committed:
Added JUnit Tests

ivorcmx committed:
Added javadocs

ivorcmx committed:
Fixed Coding Standards

ivorcmx committed:
Merge tag 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Event.java
#	src/main/java/duke/Todo.java

ivorcmx committed:
Added find function

geetengtan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,194 @@
+ Class to process the Tasklist of the chatbot
+ */
+public class TaskList {
+
```
Since all methods has it as parameter, maybe ArrayList<Task> could have been an attribute of this class for the sake of OO

geetengtan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,194 @@
+     * @return Nothing
+     *
+     */
+    public static void printTheList(ArrayList<Task> storeList) {
```
Method names can follow a standardised convention. For example, listCommand, deleteTaskCommand, unmarkCommand

geetengtan reviewed (commented):
Very nicely done. Interesting persona and interactions for the software.

ivorcmx committed:
Add checkstyle for gradle

ivorcmx committed:
Add checkstyle.xml

ivorcmx committed:
Added GUI interface

dannytayjy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+            }
+        }
+        sc.close();
+    }
```
- Missing Javadocs comments
- You might want to add line breaks to improve readability

dannytayjy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    public static String unknownCommand() {
+        return "Master, I have all the knowledge in the world but I do not recognise that command," +
+                " Please wish again";
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+
+    public static String startGoodbye() {
+        return "Rub my lamp to summon me again, Good bye for now master";
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+    public static String startGreeting() {
+        return "A very good day to you master, I'm Blue the Genie " +
+                "What do you wish for today? Your wish is my command";
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,24 @@
+
+    char type;
+
+    public Todo(String description) {
+        super(description);
+        this.type = 't';
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,24 @@
+        this.type = 't';
+    }
+
+    /**
+     *
+     * Method to convert task to String type to be printed in the task list
+     *
+     */
+    public String toString() {
+        return "[T]" + super.toString();
+    }
```
Incomplete Javadocs comments

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+     *
+     */
+    public static String unmarkCommand(String command, ArrayList<Task> storeList) {
+        int taskNumber1 = Integer.parseInt(command.substring(7));
```
It would be better if the variable name is `taskNumber` instead of `taskNumber1`

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+     *
+     * @param command command by user which specify task description
+     * @param storeList tasklist to be added to
+     * @return Nothing
```
Wrong Javadocs comments, I believe String is returned instead of nothing

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+            String str = sb.toString();
+            return str;
+        }
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+    public static String findTask(String command, ArrayList<Task> storeList) {
+        String itemToFind = command.substring(5);
+        int sizeOfList = storeList.size();
+        ArrayList<Task> tobePrinted = new ArrayList<>();
```
Variable name `tobePrinted` should be `toBePrinted`. However, `toBePrinted` is not a good name. A better suggestion will be `tasksToBePrinted`.

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+    public static String deleteTask(String command, ArrayList<Task> storeList) {
+        if (command.length() <= 7) {
+            return "Master, you wished wrongly. Remember you have to wish in this format " +
+                    "delete tasknumber. Please wish again";
```
`+` should be on the next line, just like line 47 and 48. I see this problem at other places as well.

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+            Task tobeMark = storeList.get(taskNumber - 1);
+            tobeMark.setMark();
+           return "You could have gotten me to help you. This task has been marked done" +
+                   tobeMark.toString();
```
- `+` should be on the next line
- Indentation problem

dannytayjy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,79 @@
+    /**
+     *
+     * Method to set the task to be done
+     * @return nothing
```
I believe if the method returns nothing, you can perfectly omit this line

dannytayjy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,79 @@
+     * @return nothing
+     */
+    public void setMark() {
+
```
You might want to remove this additional line break

dannytayjy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,79 @@
+    /**
+     *
+     * Method to set the task to be not done
+     * @return nothing
```
Same as above

dannytayjy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,128 @@
+                }
+            }
+        }
+    }
```
- Deep nesting here is not that bad, but it is possible to SLAP the codes
- If you do not wish to SLAP the codes, you can add line breaks to improve readability

dannytayjy commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,26 @@
+        super(description);
+        this.at = at;
+        this.type = 'e';
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public class Event extends Task {
+
+    protected String at;
```
Variable name `at` is not very ideal. `eventDateTime` would be better.

dannytayjy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+ Supports date function and description of task
+ */
+public class Deadline extends Task {
+    protected String by;
```
Variable name `by` is not very ideal. `deadlineDateTime` would be better.

dannytayjy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+        super(description);
+        this.by = by;
+        this.type = 'd';
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+        DateTimeFormatter newFormat = DateTimeFormatter.ofPattern("dd MMM yyyy");
+        String dateFormatted = date.format(newFormat);
+        return dateFormatted;
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+    public String getTime() {
+        String timeclock = by.substring(14);
+        return timeclock;
+    }
```
Missing Javadocs comments

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+            return "Master, you wished wrongly. Remember you have to wish in this format " +
+                    "deadline task /at dateofevent. Please wish again";
+        } else {
+            String newtask = command.substring(6, slash - 1);
```
Variable name `newtask` should be `newTask`

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+                    "deadline task /at dateofevent. Please wish again";
+        } else {
+            String newtask = command.substring(6, slash - 1);
+            String attime = command.substring(slash + 1);
```
Variable name `attime` should be `atTime`

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+        } else {
+            String newtask = command.substring(6, slash - 1);
+            String attime = command.substring(slash + 1);
+            Event e = new Event(newtask, attime);
```
Variable name `e` should be `event`

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+            return "Master, I have all the knowledge in the world but I do not know what you want to do," +
+                    " Please wish again in the format todo task";
+        } else {
+            String newtask = command.substring(5);
```
Same as mentioned above

dannytayjy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+                    " Please wish again in the format todo task";
+        } else {
+            String newtask = command.substring(5);
+            Todo t = new Todo(newtask);
```
Same as mentioned above

dannytayjy reviewed (commented):
I think your codes are generally good. Code quality is generally fine, except for one part that I thought might be possible to SLAP. 

Issues with Coding Standards:
- Incomplete or missing Javadocs comments
- Variable names
- `+` should be the next line
- Try and add some line breaks where appropriate to improve readability

seksek13 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,128 @@
+     * @throws IOException If path is not valid
+     */
+    public static void restoreList(ArrayList<Task> storeList) throws IOException {
+        String currentDirectory = Paths.get("Duke.java").toAbsolutePath().getParent().toString();
+        String newFilePath = currentDirectory + "/duke.txt";
```
I like the variable names here, it is clear and concise

seksek13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+                    "delete tasknumber. Please wish again";
+        } else {
+            try {
+                Task t = storeList.get(Integer.parseInt(command.substring(7)));
```
maybe it would be clearer to rename the variable "t" to something more meaningful

seksek13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+                        "deadline task /by dateofdeadline. Please wish again";
+            } else {
+                String endtime = command.substring(slash + 1);
+                Deadline d = new Deadline(newtask, endtime);
```
how about changing the variable "d" to other names which will give more information about what you are doing :)

seksek13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+                return "Master, you wished wrongly. Remember you have to wish in this format " +
+                        "deadline task /by dateofdeadline. Please wish again";
+            } else {
+                String endtime = command.substring(slash + 1);
```
the variable "endtime" should be changed to "endTime"

seksek13 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+        } else if (taskNumber1 <= 0) {
+            return "Master, I am only a genie. Please choose a number between 1 and " + storeList.size();
+        } else {
+            Task tobeUnmark = storeList.get(taskNumber1 - 1);
```
how about changing "tobeUnmark" to "taskToUnmark"?

seksek13 reviewed (commented):
There quite a few variable names that could be changed to improve clarification of the code, other than that I no other problems, good job! 👍

ivorcmx committed:
Add FXML

ivorcmx committed:
Updated MainWindow.fxml

ivorcmx committed:
Fixed code quality mistakes

ivorcmx committed:
Added Assert feature

There are several methods that rely on important assumptions to ensure
code workability.

As a step towards ensuring usability of code, let's add assert
functionality towards the important methods. Doing so will ensure
the usability of code in certain environments

ivorcmx committed:
Merge pull request #1 from ivorcmx/A-Assertions

Added Assert feature

ivorcmx committed:
Merge branch 'master' of https://github.com/ivorcmx/iP

ivorcmx committed:
Improved code quality

There were several methods which should had extracted methods to improve
on code quality as well as readability

The changes have been made to adhere to code quality conventions

ivorcmx committed:
Merge tag 'A-CodeQuality'

ivorcmx committed:
Added Update functionality

It is common for users to have typos in the description. The update
function helps to combat that problem by giving users the opportunity
to update the task description to a new String easily.

ivorcmx committed:
Change GUI

ivorcmx committed:
Updated GUI

ivorcmx committed:
Updated README.md

ivorcmx committed:
Set theme jekyll-theme-architect

ivorcmx committed:
Updated build.gradle

ivorcmx committed:
Merge branch 'master' of https://github.com/ivorcmx/iP

ivorcmx committed:
Updated README.md
